apps/storefront/src/pages/AccountSetting/index.tsx

```diff
@@ -185,7 +187,19 @@ function AccountSetting() {
   const validateEmailValue = async (emailValue: string) => {
     if (customer.emailAddress === trim(emailValue)) return true;
 
-    if (!isValidUserType) {
+    const fn = !isBCUser ? checkUserEmail : checkUserBCEmail;
+    const key = !isBCUser ? 'userEmailCheck' : 'customerEmailCheck';
+
+    const {
+      [key]: { userType },
+    }: CustomFieldItems = await fn({
+      email: emailValue,
+      channelId,
+    });
+
+    const isValid = !isBCUser ? [1].includes(userType) : ![2].includes(userType);
+
+    if (!isValid) {
```

Comment:
@icatalina made a refactor about how to fetch data from graphql

Please take a look and take it as base for this kind of validation to decide which fetch should be request -> https://github.com/bigcommerce/b2b-buyer-portal/pull/30/files#diff-7ecfe89a8df11fedd9c965ef7ea2965c47158903d683ecdb90ab29a4722653ef